### PR TITLE
ci: update static artifact actions to v4

### PR DESCRIPTION
**Pull request by vinceAmstoutz**
Completes #1255

### CODE REVIEW

**Review by withinboredom — APPROVED** [review]

File: .github/workflows/static.yaml [hunk 1]
```diff
@@ -173,9 +179,9 @@ jobs:
       -
         name: Upload artifact
         if: ${{ !fromJson(needs.prepare.outputs.push) }}
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: frankenphp-linux-${{ matrix.platform == 'linux/amd64' && 'x86_64' || 'aarch64' }}${{ matrix.debug && '-debug' || '' }}${{ matrix.mimalloc && '-mimalloc' || '' }}
+          name: frankenphp-linux-${{ matrix.platform == 'linux/amd64' && 'x86_64' || 'aarch64' }}${{ matrix.debug && '-debug' || '' }}${{ matrix.mimalloc && '-mimalloc' || '' }}-${{ steps.prepare.outputs.sanitized_platform }}
```
Comment (dunglas) [inline .github/workflows/static.yaml:184]: This should be reverted, it's a classic upload, not a trick to pass metadata to another workflow. Also, the platform is already in the name.

File: .github/workflows/static.yaml [hunk 2]
```diff
@@ -259,6 +266,12 @@ jobs:
     env:
       HOMEBREW_NO_AUTO_UPDATE: 1
     steps:
+      -
+        name: Prepare
+        id: prepare
+        run: |
+          platform=${{ matrix.platform }}
+          echo "sanitized_platform=${platform//\//-}" >> "${GITHUB_OUTPUT}"
```
Comment (dunglas) [inline .github/workflows/static.yaml:274]: It's not necessary here, we don't use Docker for Mac builds.

File: .github/workflows/static.yaml [hunk 3]
```diff
@@ -297,7 +310,7 @@ jobs:
       -
         name: Upload artifact
         if: github.ref_type == 'branch'
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: frankenphp-mac-${{ matrix.platform }}
+          name: frankenphp-mac-${{ matrix.platform }}-${{ steps.prepare.outputs.sanitized_platform }}
```
Comment (dunglas) [inline .github/workflows/static.yaml:315]: To revert, it's a classic upload.

**Review by dunglas — CHANGES_REQUESTED** [review]

File: .github/workflows/static.yaml [hunk 4]
```diff
@@ -154,7 +154,7 @@ jobs:
       -
         name: Upload metadata
         if: fromJson(needs.prepare.outputs.push) && !matrix.debug && !matrix.mimalloc
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: metadata-static-builder
```
Comment (dunglas) [inline .github/workflows/static.yaml:159]: Here we need the sanitized platform

**Review by dunglas — CHANGES_REQUESTED** [review]

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Thanks!

**Pull request merged**